### PR TITLE
new command `date`

### PR DESCRIPTION
**Pull request by maxx0911**
This command will give users another oppurtunity to increase their lovescore with their partner and is overall another fun aspect to the marriage side of the bot. The number range for `num` could be fine-tuned to be a bit higher, but let me know what you think.

### CODE REVIEW

**Comment by Gelbpunkt** [comment]
@maxx0911 look at my comments please ☺️

**Comment by maxx0911** [comment]
2/3 changed, need some help with `user_on_cooldown` and how to properly use that as I found no info on it.

**Comment by maxx0911** [comment]
Nevermind, I have just found how to use it. All three requested changes are made.

File: cogs/marriage.py [hunk 1]
```diff
@@ -236,6 +236,41 @@ def reactioncheck(reaction, user):
...
+
+            if marriage == 0:
+                return await ctx.send("You are not married.")
+
+            await conn.execute(
+                'UPDATE profile SET lovescore=lovescore+$1 WHERE "user"=$2;',
+                num,
+                marriage,
+            )
+        partner = await rpgtools.lookup(self.bot, marriage)
```
Comment (Gelbpunkt) [inline cogs/marriage.py]: Here, use `partner = await bot.get_user_global(marriage)`

File: cogs/marriage.py [hunk 2]
```diff
@@ -236,6 +236,41 @@ def reactioncheck(reaction, user):
...
+        scenario = random.choice(
+            [
+                f"You and {partner} went on a nice candlelit dinner.",
```
Comment (Gelbpunkt) [inline cogs/marriage.py]: This and all below: `partner.mention`

File: cogs/marriage.py [hunk 3]
```diff
@@ -236,6 +236,41 @@ def reactioncheck(reaction, user):
...
+    @user_cooldown(43200)
+    async def _date(self, ctx):
+        """Take your loved one on a date to increase your lovescore."""
+        num = random.randint(1, 15) * 10
+        async with self.bot.pool.acquire() as conn:
+            marriage = await conn.fetchval(
+                'SELECT marriage FROM profile WHERE "user"=$1;', ctx.author.id
+            )
+
+            if marriage == 0:
```
Comment (Gelbpunkt) [inline cogs/marriage.py]: `if not marriage`

**Comment by maxx0911** [comment]
There we go, changes made. I'm not exactly agreeing that the partner should be mentioned though, as there is no message check, like with `child`, which a user should be notified of. Especially if that command gets used in direct message (to sort of surprise the partner), just the name would be better.

File: cogs/marriage.py [hunk 4]
```diff
@@ -236,6 +236,41 @@ def reactioncheck(reaction, user):
...
         )
+        
+    @has_char()
+    @commands.command(name="date")
+    @user_cooldown(43200)
+    async def _date(self, ctx):
+        """Take your loved one on a date to increase your lovescore."""
+        num = random.randint(1, 15) * 10
+        async with self.bot.pool.acquire() as conn:
+            marriage = await conn.fetchval(
```
Comment (Gelbpunkt) [inline cogs/marriage.py]: `ctx.character_data["marriage"]` already exists after the `has_char`, no need to fetch it from the database once more. That's the last thing before merging, I guess

Comment (maxx0911) [reply cogs/marriage.py]: alright, changed

**Comment by maxx0911** [comment]
Should now be ready to merge now.

**Comment by Gelbpunkt** [comment]
You have to replace it in the bot.get_user_global, too. Don't forget the `if not ctx.character_data["marriage"]` to ensure they're married (you removed it)

**Comment by Gelbpunkt** [comment]
Still missing the `if not marriage`

**Comment by maxx0911** [comment]
my bad, my bad... sorrry...

**Pull request merged**